lib/pure/future.nim

```diff
@@ -197,4 +197,32 @@ macro dump*(x: typed): untyped =
   let s = x.toStrLit
   let r = quote do:
     debugEcho `s`, " = ", `x`
-  return r
\ No newline at end of file
+  return r
+
+macro asArray*(targetType: typedesc, values: typed): untyped =
+  ## applies a type conversion to each of the elements in the specified
+  ## array literal. Each element is converted to the ``targetType`` type..
+  ##
+  ## Example:
+  ##
+  ## .. code-block::
+  ##   let x = asArray(int, [0.1, 1.2, 2.3, 3.4])
+  ##   doAssert x is array[4, int]
+  ##
+  ## Short notation for:
+  ##
+  ## .. code-block::
+  ##   let x = [(0.1).int, (1.2).int, (2.3).int, (3.4).int]
+  let tNode = getType(targetType)[1]
```

Comment:
Don't use ``getType`` here. Use a purely syntactic macro.

Reply:
@Araq sorry, I have no idea what you mean. Don't I need `getType`... else I got `targetType` in the produced code. I am **very probably** doing something fundamentally wrong, but I don't understand how else I should do this :(

Reply:
maybe you mean like I did here: https://github.com/nim-lang/Nim/pull/6640/commits/31ace8ad297731cf2f9f43f995a3cc794b3d77fc ?